modules/core/keeper/msg_server.go

```diff
@@ -763,3 +774,23 @@ func (k Keeper) ChannelUpgradeTimeout(goCtx context.Context, msg *channeltypes.M
 func (k Keeper) ChannelUpgradeCancel(goCtx context.Context, msg *channeltypes.MsgChannelUpgradeCancel) (*channeltypes.MsgChannelUpgradeCancelResponse, error) {
 	return nil, nil
 }
+
+// extractUpgradeFromMessage returns the proposed upgrade from the provided MsgChannelUpgradeInit message.
+func (k Keeper) extractUpgradeFromMessage(ctx sdk.Context, msg *channeltypes.MsgChannelUpgradeInit) (channeltypes.Upgrade, error) {
+	seq, found := k.ChannelKeeper.GetNextSequenceSend(ctx, msg.PortId, msg.ChannelId)
+	if !found {
+		return channeltypes.Upgrade{}, channeltypes.ErrSequenceSendNotFound
```

Comment:
I'm unsure if this is the desired behaviour. If we don't want to error here, what value should be set for `LatestSequenceSend`, 0?

Reply:
I think if next sequence send is not found then we should error